### DISCUSSION TITLE
NC-AIO behind proxy (NPM) => more than one HSTS header

### DISCUSSION THREAD

**Discussion by tullsta2** [nextcloud/all-in-one#3652] [Questions] [answered]
Hello,

I've installed nextcloud AIO following the documentation and run it behind a reverse proxy - in my case Nginx-Proxy-Manager (NPM).
when checking the site using Mozilla Observatory or SSLlabs, I get an HSTS-error (Mozilla) and an error saying "npm Server provided more than one HSTS header" (SSLlabs).

Now, when using NPM I can en- or disable HSTS. The docu says one should enable, but the error disappears when I disable this in NPM.
So there must be a different source for a 2nd max-age entry:
NPM inserts one (add_header Strict-Transport-Security "max-age=63072000...)
I went though my setup in detail, but I'm still wondering who creates that 2nd max-age entry (max-age=31536000).

Maybe someone can help out here.

Thanks and kind regards - tullsta

**Comment by szaimen** [comment, marked as answer]
Hi, indeed the built-in caddy already adds the header. So it is not needed to enable it in NPM. Would you mind updating the screenshot in our docs?

**Reply by tullsta2** [reply]
thanks for the fast reply - I will update later today.

Kind regards - tullsta

**Comment by tullsta2** [comment]
Hello - unfortunately did not manage to upload/change the screenshot.
I added a note and here the accompanying screenshot:
![Screenshot 2023-10-31 234449](https://github.com/nextcloud/all-in-one/assets/24786786/fecbb5ef-d2f4-4e0f-bc4b-82207e2c2809)

Thanks + kind regards - tullsta

**Reply by szaimen** [reply]
Done in https://github.com/nextcloud/all-in-one/commit/e6b6ec4cfd0a959697aa369bb9679ebc696b3ad5
